File: python/2_fib.py


def fib(n):
  fibs = []
  fibs.append(0)
  fibs.append(1)
  for i in xrange(2,n):
    fibs.append(fibs[i - 2] + fibs[i-1])

  return fibs

series = fib(34)
series = filter(lambda x: x %2 == 0, series)

print sum(series)

File: javascript/2_fib.js


function fib(n){

  var list = [];
  list.push(0);
  list.push(1);
  for(var i = 2; i < n; i++){
    list.push(list[i-1] + list[i-2]);
  }
  return list;
}

var list = fib(34);
var new_list = list.filter(function(x){return x % 2 == 0;});
var sum = new_list.reduce(function(pv, cv){ return pv+cv;}, 0);
console.log(sum);

